File: src/leetcode/count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.js
/**
 * Leetcode #2006
 * Difficulty: Easy
 * https://leetcode.com/problems/count-number-of-pairs-with-absolute-difference-k/
 */
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
export const countKDifference = (nums, k) => {
  nums.sort((a, b) => a - b);

  let ans = 0;

  for (let i = 0; i < nums.length; i += 1) {
    for (let j = i + 1; j < nums.length; j += 1) {
      if (nums[j] - nums[i] === k) {
        ans += 1;
      }
    }
  }

  return ans;
};

// const countKDifference = (nums, k) => {
//   let ans = 0;
//
//   for (let i = 0; i < nums.length; i += 1) {
//     for (let j = i + 1; j < nums.length; j += 1) {
//       if (Math.abs(nums[j] - nums[i]) === k) {
//         ans += 1;
//       }
//     }
//   }
//
//   return ans;
// };

File: src/leetcode/count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.rs
impl Solution {
    pub fn count_k_difference(nums: Vec<i32>, k: i32) -> i32 {
        let mut ans = 0;
        let len = nums.len();

        for i in 0..len {
            for j in i + 1..len {
                if i32::abs(nums[j] - nums[i]) == k {
                    ans += 1;
                }
            }
        }

        ans
    }
}

// impl Solution {
//     pub fn count_k_difference(mut nums: Vec<i32>, k: i32) -> i32 {
//         let mut ans = 0;
//         let len = nums.len();
//         nums.sort();
//
//         for i in 0..len {
//             for j in i + 1..len {
//                 if (nums[j] - nums[i] == k) {
//                     ans += 1;
//                 }
//             }
//         }
//
//         ans
//     }
// }
//
